ophyd/status.py

```diff
@@ -38,7 +43,7 @@ def __init__(self, *, timeout=None, settle_time=None, done=False,
                  success=False):
         super().__init__()
         self._lock = RLock()
-        self._cb = None
+        self._callbacks = set()
```

Comment:
Probably better to use a dequeue or list so we can make guarantees about order the callbacks are run in.  I am not sure I _like_ it being used that way, but cpython makes these sort of garuntees on `atexit` and `weakref` callbacks.

also I am channeling @klauer